Mage.Sets/src/mage/cards/l/LiberatedLivestock.java

```diff
@@ -0,0 +1,121 @@
+import mage.abilities.common.DiesSourceTriggeredAbility;
+import mage.abilities.effects.OneShotEffect;
+import mage.cards.*;
+import mage.constants.Outcome;
+import mage.constants.SubType;
+import mage.constants.CardType;
+import mage.constants.Zone;
+import mage.filter.FilterCard;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+import mage.game.permanent.token.*;
+import mage.players.Player;
+import mage.target.TargetCard;
+import mage.target.common.TargetCardInHand;
+import mage.target.common.TargetCardInYourGraveyard;
+
+/**
+ *
+ * @author Codermann63
+ */
+public final class LiberatedLivestock extends CardImpl {
+
+    public LiberatedLivestock(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{5}{W}");
+        
+        this.subtype.add(SubType.CAT);
+        this.subtype.add(SubType.BIRD);
+        this.subtype.add(SubType.OX);
+        this.power = new MageInt(4);
+        this.toughness = new MageInt(6);
+
+
+
+        // When Liberated Livestock dies, create a 1/1 white Cat creature token with lifelink,
+        // a 1/1 white Bird creature token with flying,
+        // and a 2/4 white Ox creature token.
+        // For each of those tokens, you may put an Aura card from your hand and/or graveyard onto the battlefield attached to it.
+        this.addAbility(new DiesSourceTriggeredAbility(new LiberatedLivestockEffect()));
+    }
+
+    private LiberatedLivestock(final LiberatedLivestock card) {
+        super(card);
+    }
+
+    @Override
+    public LiberatedLivestock copy() {
+        return new LiberatedLivestock(this);
+    }
+}
+
+class LiberatedLivestockEffect extends OneShotEffect {
+
+    LiberatedLivestockEffect() {
+        super(Outcome.PutCreatureInPlay);
+        staticText = "create a 1/1 white Cat creature token with lifelink, a 1/1 white Bird creature token with flying, and a 2/4 white Ox creature token. For each of those tokens, you may put an Aura card from your hand and/or graveyard onto the battlefield attached to it.";
+    }
+
+    private LiberatedLivestockEffect(final LiberatedLivestockEffect effect) {super(effect);}
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player controller = game.getPlayer((source.getControllerId()));
+        if (controller == null){
+            return false;
+        }
+        Token cat = new CatToken2();
+        Token bird = new BirdToken();
+        Token ox = new OxToken();
+
+        // pause to get token onto the battlefield all the same time
+        game.getState().pause();
+        cat.putOntoBattlefield(1, game, source, source.getControllerId());
+        bird.putOntoBattlefield(1, game, source, source.getControllerId());
+        ox.putOntoBattlefield(1, game, source, source.getControllerId());
+        game.getState().resume();
+
+        // Setup for choosing and attaching auras to tokens
+        List<Token> tokens = Arrays.asList(cat, bird, ox);
```

Comment:
can simplify by defining this array first and then iterating through it for the .putOntoBattlefield(...)